Failure in system_tests_sasl_plain

```
=========================================================================== FAILURES ===========================================================================
26: _________________________________________________________ RouterTestPlainSasl.test_skstat_connect_sasl _________________________________________________________
26: 
26: self = <tests.system_tests_sasl_plain.RouterTestPlainSasl testMethod=test_skstat_connect_sasl>
26: 
26:     @unittest.skipIf(not SASL.extended(), "Cyrus library not available. skipping test")
26:     def test_skstat_connect_sasl(self):
26:         """
26:         Make skstat use sasl plain authentication.
26:         """
26:     
26:         p = self.popen(
26:             ['skstat', '-b', str(self.routers[0].addresses[2]), '-c', '--sasl-mechanisms=PLAIN',
26:              '--sasl-username=test@domain.com', '--sasl-***'],
26:             name='skstat-' + self.id(), stdout=PIPE, expect=Process.EXIT_OK,
26:             universal_newlines=True)
26:     
26:         out = p.communicate()[0]
26:         assert p.returncode == 0, \
26:             "skstat exit status %s, output:\n%s" % (p.returncode, out)
26:     
26:         split_list = out.split()
26:     
26:         # There will be 1 inter-router connection and 1 skstat client connection that
26:         # have authenticated using SASL PLAIN. And also the number of inter-router-data
26:         # connections that are determined by the worker thread count.
26:         n_total_connections = worker_threads_to_data_connections(n_worker_threads) + 1 + 1
26: >       self.assertEqual(n_total_connections, split_list.count("test@domain.com(PLAIN)"))
26: E       AssertionError: 3 != 2
26: 
26: /home/runner/work/skupper-router/skupper-router/skupper-router/tests/system_tests_sasl_plain.py:367: AssertionError
26: __________________________________________________ RouterTestPlainSasl.test_skstat_connect_sasl_password_file __________________________________________________
26: 
26: self = <tests.system_tests_sasl_plain.RouterTestPlainSasl testMethod=test_skstat_connect_sasl_password_file>
26: 
26:     @unittest.skipIf(not SASL.extended(), "Cyrus library not available. skipping test")
26:     def test_skstat_connect_sasl_password_file(self):
26:         """
26:         Make skstat use sasl plain authentication with client password specified in a file.
26:         """
26:         password_file = os.getcwd() + '/sasl-client-password-file.txt'
26:         # Create a SASL configuration file.
26:         with open(password_file, 'w') as sasl_client_password_file:
26:             sasl_client_password_file.write("password")
26:     
26:         sasl_client_password_file.close()
26:     
26:         p = self.popen(
26:             ['skstat', '-b', str(self.routers[0].addresses[2]), '-c', '--sasl-mechanisms=PLAIN',
26:              '--sasl-username=test@domain.com', '--sasl-password-file=' + password_file],
26:             name='skstat-' + self.id(), stdout=PIPE, expect=Process.EXIT_OK,
26:             universal_newlines=True)
26:     
26:         out = p.communicate()[0]
26:         assert p.returncode == 0, \
26:             "skstat exit status %s, output:\n%s" % (p.returncode, out)
26:     
26:         split_list = out.split()
26:     
26:         # There will be 1 inter-router connection and 1 skstat client connection that
26:         # have authenticated using SASL PLAIN. And also the number of inter-router-data
26:         # connections that are determined by the worker thread count.
26:         n_total_connections = worker_threads_to_data_connections(n_worker_threads) + 1 + 1
26: >       self.assertEqual(n_total_connections, split_list.count("test@domain.com(PLAIN)"))
26: E       AssertionError: 3 != 2
26: 
```

https://github.com/skupperproject/skupper-router/actions/runs/9843223952/job/27173964840?pr=1544#step:34:2535

--- a/tests/system_tests_sasl_plain.py
+++ b/tests/system_tests_sasl_plain.py
@@ -347,26 +347,30 @@ class RouterTestPlainSasl(RouterTestPlainSaslCommon):
         """
         Make skstat use sasl plain authentication.
         """
-
-        p = self.popen(
-            ['skstat', '-b', str(self.routers[0].addresses[2]), '-c', '--sasl-mechanisms=PLAIN',
-             '--sasl-username=test@domain.com', '--sasl-password=password'],
-            name='skstat-' + self.id(), stdout=PIPE, expect=Process.EXIT_OK,
-            universal_newlines=True)
-
-        out = p.communicate()[0]
-        assert p.returncode == 0, \
-            "skstat exit status %s, output:\n%s" % (p.returncode, out)
-
-        split_list = out.split()
-
         # There will be 1 inter-router connection and 1 skstat client connection that
         # have authenticated using SASL PLAIN. And also the number of inter-router-data
         # connections that are determined by the worker thread count.
         n_total_connections = worker_threads_to_data_connections(n_worker_threads) + 1 + 1
-        self.assertEqual(n_total_connections, split_list.count("test@domain.com(PLAIN)"))
-        self.assertEqual(1, split_list.count("inter-router"))
-        self.assertEqual(1, split_list.count("normal"))
+
+        def check_sasl_on_connections():
+            p = self.popen(
+                ['skstat', '-b', str(self.routers[0].addresses[2]), '-c', '--sasl-mechanisms=PLAIN',
+                 '--sasl-username=test@domain.com', '--sasl-password=password'],
+                name='skstat-' + self.id(), stdout=PIPE, expect=Process.EXIT_OK,
+                universal_newlines=True)
+
+            out = p.communicate()[0]
+            assert p.returncode == 0, \
+                "skstat exit status %s, output:\n%s" % (p.returncode, out)
+
+            split_list = out.split()
+
+            self.assertEqual(n_total_connections, split_list.count("test@domain.com(PLAIN)"))
+            self.assertEqual(1, split_list.count("inter-router"))
+            self.assertEqual(1, split_list.count("normal"))
+
+        # Sometimes the router needs time to establish the connections, especially on a slow CI system.
+        retry_assertion(check_sasl_on_connections, delay=1)
 
     @unittest.skipIf(not SASL.extended(), "Cyrus library not available. skipping test")
     def test_skstat_connect_sasl_password_file(self):
@@ -377,28 +381,31 @@ class RouterTestPlainSasl(RouterTestPlainSaslCommon):
         # Create a SASL configuration file.
         with open(password_file, 'w') as sasl_client_password_file:
             sasl_client_password_file.write("password")
-
         sasl_client_password_file.close()
-
-        p = self.popen(
-            ['skstat', '-b', str(self.routers[0].addresses[2]), '-c', '--sasl-mechanisms=PLAIN',
-             '--sasl-username=test@domain.com', '--sasl-password-file=' + password_file],
-            name='skstat-' + self.id(), stdout=PIPE, expect=Process.EXIT_OK,
-            universal_newlines=True)
-
-        out = p.communicate()[0]
-        assert p.returncode == 0, \
-            "skstat exit status %s, output:\n%s" % (p.returncode, out)
-
-        split_list = out.split()
 
         # There will be 1 inter-router connection and 1 skstat client connection that
         # have authenticated using SASL PLAIN. And also the number of inter-router-data
         # connections that are determined by the worker thread count.
         n_total_connections = worker_threads_to_data_connections(n_worker_threads) + 1 + 1
-        self.assertEqual(n_total_connections, split_list.count("test@domain.com(PLAIN)"))
-        self.assertEqual(1, split_list.count("inter-router"))
-        self.assertEqual(1, split_list.count("normal"))
+
+        def check_sasl_on_connections():
+            p = self.popen(
+                ['skstat', '-b', str(self.routers[0].addresses[2]), '-c', '--sasl-mechanisms=PLAIN',
+                 '--sasl-username=test@domain.com', '--sasl-password-file=' + password_file],
+                name='skstat-' + self.id(), stdout=PIPE, expect=Process.EXIT_OK,
+                universal_newlines=True)
+
+            out = p.communicate()[0]
+            assert p.returncode == 0, \
+                "skstat exit status %s, output:\n%s" % (p.returncode, out)
+
+            split_list = out.split()
+            self.assertEqual(n_total_connections, split_list.count("test@domain.com(PLAIN)"))
+            self.assertEqual(1, split_list.count("inter-router"))
+            self.assertEqual(1, split_list.count("normal"))
+
+        # Sometimes the router needs time to establish the connections, especially on a slow CI system.
+        retry_assertion(check_sasl_on_connections, delay=1)
 
 
 class RouterTestPlainSaslOverSsl(RouterTestPlainSaslCommon):

